SAAS-68 Fix issues with go logging facility

kostrahb opened the pull request.

Shimmi commented on _examples/init/config.json:
```diff
@@ -1,5 +1,7 @@
 {
 	"logging": {
+		"env": "dev",
+		"project": "connect",
```
jelikož je tohle obecná logovací utilita, použij prosín neutrální název. E.g. "My project name"

Shimmi commented on handlers/init/Initialization.go:
```diff
@@ -1,71 +1,72 @@
 
 	var handlers []log.Handler
-	if(h["json"] != nil) {
+	if h["json"] != nil {
```
a ještě vrať ty závorky :)

kostrahb committed:
[SAAS-68] Fixed thread safety

Calling Exit() from multiple threads concurently now executes each function exactly once.

kostrahb committed:
[SAAS-68] Fixed hook logic

If multiple hooks are executed on single field their effect stacks and also if no hook is executed the value is passed as is.
